### PR TITLE
Add VisionOS and macOS support

### PR DESCRIPTION
**Pull request by philippzagar**
# Add VisionOS and macOS support

## :recycle: Current situation & Problem
Currently, SpeziChat only supports iOS platforms.

watchOS and tvOS both don't have the `Speech` framework (important for SpeziSpeech), sadly preventing us from adding platform support.

## :gear: Release Notes 
- Add VisionOS and macOS support

## :books: Documentation
--

## :white_check_mark: Testing
CI steps to ensure building & launching on the respective platforms.

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by philippzagar** [comment]
@PSchmiedmayer Please adjust the merge checks accordingly to the new CI setup, so that we can merge the PR after the SpeziSpeech PR is merged 🚀

**Comment by PSchmiedmayer** [comment]
@philippzagar I merged the changes in SpeziSpeech and adjusted the requirements here. We just need to fix the visionOS test error here before we can merge the PR 👍

File: Package.swift [hunk 1]
```diff
@@ -15,13 +15,15 @@ let package = Package(
...
+        .iOS(.v17),
+        .visionOS(.v1),
+        .macOS(.v14)
     ],
     products: [
         .library(name: "SpeziChat", targets: ["SpeziChat"])
     ],
     dependencies: [
-        .package(url: "https://github.com/StanfordSpezi/SpeziSpeech", from: "1.0.0")
+        .package(url: "https://github.com/StanfordSpezi/SpeziSpeech", branch: "feat/platforms")
```
Comment (PSchmiedmayer) [inline Package.swift:26]: TODO: Update to the release of Spezi Speech.

File: Sources/SpeziChat/ChatView+ShareSheet.swift [hunk 2]
```diff
@@ -43,4 +40,41 @@ extension ChatView {
...
+        }
+    }
+    #endif
+    
+    
+    /// Constructs the temporary file path for the exported chat file.
+    ///
+    /// - Parameters:
+    ///    - sharedItemType: The shared item type, therefore defining the file extension.
+    static func temporaryExportFilePath(sharedItemType: ChatExportFormat) -> URL {
```
Comment (PSchmiedmayer) [inline Sources/SpeziChat/ChatView+ShareSheet.swift:69]: Nice to share this between operating systems.

File: Sources/SpeziChat/MessageInputView.swift [hunk 3]
```diff
@@ -50,25 +50,46 @@ public struct MessageInputView: View {
...
+                        #endif
                 }
                 .lineLimit(1...5)
+                #if os(visionOS) && TEST
+                // Workaround on visionOS as UI tests are not able to properly set focus on `TextField`
+                .focused($inputFieldFocus)
+                .onTapGesture {
+                    inputFieldFocus = true
+                }
+                #endif
```
Comment (PSchmiedmayer) [inline Sources/SpeziChat/MessageInputView.swift:92]: This is unfortunate ... do we have an idea where this is stemming from? Maybe the background or something?

File: Sources/SpeziChat/MessageInputView.swift [hunk 4]
```diff
@@ -96,8 +117,14 @@ public struct MessageInputView: View {
                 }
             }
             .messageInputViewHeight(messageViewHeight)
+            #if os(macOS)
+            .onSubmit {
+                sendMessageButtonPressed()
+            }
+            #endif
```
Comment (PSchmiedmayer) [inline Sources/SpeziChat/MessageInputView.swift:124]: Does it even make sense to also offer this on iOS, e.g., when the user has a connected keyboard? Or is that already covered?

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for the improvements; let's discuss some of the comments in our meeting tomorrow 🚀

**Pull request merged**